examples/custom-client-cxx/cli.cpp

```diff
@@ -0,0 +1,365 @@
+  } catch (const std::exception& exc) {
+    LOG_ERROR << "Failed to initialize the client: " << exc.what();
+    return EXIT_FAILURE;
+  }
+
+  try {
+    CheckInResult res = !is_local_update?
+                            client->CheckIn():
+                            client->CheckInLocal(&local_update_source);
+    if (res.status != CheckInResult::Status::Ok && res.status != CheckInResult::Status::OkCached) {
+      LOG_WARNING << "Unable to update latest metadata";
+      return EXIT_FAILURE;
+    }
+
+    LOG_INFO << "Found " << res.Targets().size() << " targets";
+    for (const auto &target : res.Targets()) {
+      LOG_INFO << target.Name();
+    }
+  } catch (const std::exception &exc) {
+    LOG_ERROR << "Failed to find or update Target: " << exc.what();
+    return EXIT_FAILURE;
+  }
+  return EXIT_SUCCESS;
+}
+
+int cmd_pull(std::string target_name, std::string local_repo_path) {
+  boost::log::core::get()->set_filter(boost::log::trivial::severity >= boost::log::trivial::info);
+  bool is_local_update{false};
+  LocalUpdateSource local_update_source;
+
+  std::vector<boost::filesystem::path> cfg_dirs;
+  auto env{boost::this_process::environment()};
+  if (env.end() != env.find("AKLITE_CONFIG_DIR")) {
+    cfg_dirs.emplace_back(env.get("AKLITE_CONFIG_DIR"));
+  } else {
+    cfg_dirs = AkliteClient::CONFIG_DIRS;
+  }
+
+  if (!local_repo_path.empty()) {
+    local_update_source = {
+      local_repo_path + "/tuf",
+      local_repo_path + "/ostree_repo",
+      local_repo_path + "/apps"
+    };
+    is_local_update = true;
+    LOG_INFO << "Offline mode. Updates path=" << local_repo_path;
+  } else {
+    LOG_INFO << "Online mode";
+  }
+
+  std::unique_ptr<AkliteClient> client;
+  try {
+    client = std::make_unique<AkliteClient>(cfg_dirs);
+  } catch (const std::exception& exc) {
+    LOG_ERROR << "Failed to initialize the client: " << exc.what();
+    return EXIT_FAILURE;
+  }
+
+  try {
+    auto current = client->GetCurrent();
+
+    CheckInResult res = !is_local_update?
+                            client->CheckIn():
+                            client->CheckInLocal(&local_update_source);
+    if (res.status != CheckInResult::Status::Ok && res.status != CheckInResult::Status::OkCached) {
+      LOG_WARNING << "Unable to update latest metadata";
+      return EXIT_FAILURE;
+    }
+
+    aklite::tuf::TufTarget selected_target;
+    if (target_name.empty()) {
+      auto latest = res.GetLatest();
+      if (client->IsRollback(latest)) {
+        LOG_INFO << "Latest Target is marked for causing a rollback and won't be installed: " << latest.Name();
+      } else {
+        LOG_INFO << "Found Latest Target: " << latest.Name();
+        selected_target = latest;
+      }
+    } else {
+      for (auto const &target: res.Targets()) {
+        if (target.Name() == target_name) {
+          selected_target = target;
+          break;
+        }
+        LOG_ERROR << "Target " << target_name << " not found or is incomplete";
```

Comment:
I think it should be `continue` here so a user can install a specified target `target_name` if it is in the list of available targets (`res.Targets()`).